### PR TITLE
13021 - Changes for Special Resolution

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:* /bcgov/entity#13021
*Description of changes:*

Move over association-types.ts from business-edit-ui.
Enhance alteration-interface.ts.
Change import for business-lookup-interfaces.ts to bcrs-shared-components, this works nicer when using npx link.
Add in change-of-name-interface.ts, person-interface.ts, special-resolution-interface.ts.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/interfaces/business-lookup-interfaces.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { CorpTypeCd, EntityStates } from '@/enums'
+import { CorpTypeCd, EntityStates } from '@bcrs-shared-components/enums'
```
Comment (seeker25) [inline src/interfaces/business-lookup-interfaces.ts:1]: `npx link` had issues with @/enum it wouldn't resolve easily, this syntax was used in another file.

File: src/interfaces/special-resolution-interface.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import { PersonIF } from './person-interface'
+
+export interface SpecialResolutionIF {
+  meetingDate?: string
+  resolution: string
+  resolutionDate?: string
+  signingDate?: string
+  signatory?: PersonIF
+  resolutionConfirmed?: boolean
```
Comment (seeker25) [inline src/interfaces/special-resolution-interface.ts:14]: Included here: https://github.com/bcgov/business-schemas/pull/107

File: src/interfaces/change-of-name-interface.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/**
+ * A filing's business object from the API. See:
+ * https://github.com/bcgov/business-schemas/blob/master/src/registry_schemas/schemas/change_of_name.json
+ * This requires either nameRequest or legalName:
+*   "anyOf": [
```
Comment (seeker25) [inline src/interfaces/change-of-name-interface.ts:5]: Not sure if this should be oneOf instead?

**Comment by seeker25** [comment]
@severinbeauvais @lewischenstudio @lambert-alex I don't have permissions on this repo, but could you review this?

File: src/enums/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bcrs-shared-components/enums",
-  "version": "1.0.19",
+  "version": "1.0.20",
```
Comment (severinbeauvais) [inline src/enums/package.json:3]: Don't change this. The Lerna command in step 7 does this for you.

Please ensure you follow exactly the steps in the [README](https://github.com/bcgov/bcrs-shared-components) otherwise the versions can get messed, the branches can get messed up, or StoryBook can get messed up.

File: src/interfaces/court-order-interface.ts [hunk 5]
```diff
@@ -2,8 +2,9 @@
  * See:
  * https://github.com/bcgov/business-schemas/blob/master/src/registry_schemas/schemas/court_order.json
  */
-export interface CourtOrderIF{
+export interface CourtOrderIF {
   fileNumber: string
   orderDate: string
-  effectOfOrder: string
+  effectOfOrder: string,
```
Comment (severinbeauvais) [inline src/interfaces/court-order-interface.ts:8]: If you're in here again, remove the trailing comment. It's not needed and generally our code style is to leave it out so it looks cleaner.

File: src/interfaces/package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bcrs-shared-components/interfaces",
-  "version": "1.0.39",
+  "version": "1.0.40",
```
Comment (severinbeauvais) [inline src/interfaces/package.json:3]: Revert this change.

File: src/interfaces/person-interface.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+export interface PersonIF {
+  givenName?: string
+  familyName?: string
+  additionalName?: string
+  middleInitial?: string
+  email?: string
+  [propName: string]: any // excess properties
+}
+
+export const EmptyPersonIF: PersonIF = {
```
Comment (severinbeauvais) [inline src/interfaces/person-interface.ts:15]: This is an object, not an IF. Please rename it.

File: src/interfaces/person-interface.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  givenName: '',
+  familyName: '',
+  additionalName: null
```
Comment (severinbeauvais) [inline src/interfaces/person-interface.ts:18]: This is probably OK but watch out for the following:
- you're defining some optional properties
- you're inconsistently defining some string properties as empty strings, and other as null
- the remaining properties are undefined, meaning that changes to them will not trigger reactivity 

What to define in this Empty Person object depends on how it's used, which is subject not objective, which is not ideal. I'd like to define a best practise for such empty objects. Any thoughts?

Obviously, required properties should be defined. There aren't any here.

Should all optional properties be defined?

Should primitive types (eg, string, number) be assigned empty values or null?

File: src/interfaces/special-resolution-interface.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+export interface SpecialResolutionIF {
+  meetingDate?: string
+  resolution: string
+  resolutionDate?: string
+  signingDate?: string
+  signatory?: PersonIF
+  // Not in API schema, but used in UI
+  resolutionConfirmed?: boolean
+  [propName: string]: any // excess properties
```
Comment (severinbeauvais) [inline src/interfaces/special-resolution-interface.ts:16]: Again, I prefer to get rid of these and fix the code/IF instead.

File: src/interfaces/person-interface.ts (hunk 8 above)
Comment (seeker25) [reply src/interfaces/person-interface.ts:18]: I think I'll leave this out of `bcrs-shared-components`

**Review by severinbeauvais — APPROVED** [review]
Looks OK.

Some of the code things I commented on may have been carried over from existing code, but please fix anyway.

As per the readme, you also need to rebuild Storybook and check it its changes.

File: src/interfaces/court-order-interface.ts (hunk 5 above)
Comment (seeker25) [reply src/interfaces/court-order-interface.ts:8]: Whoops, good eye

File: src/interfaces/person-interface.ts (hunk 7 above)
Comment (severinbeauvais) [reply src/interfaces/person-interface.ts]: Object has been removed. This conversation is now obsolete.

File: src/interfaces/person-interface.ts (hunk 8 above)
Comment (severinbeauvais) [reply src/interfaces/person-interface.ts]: Object has been removed. This conversation is now obsolete.

**Comment by severinbeauvais** [comment]
@seeker25 Please rebuild and commit the changed Storybook files as per my comment above.

Note that only project admins can complete steps 14 and 15 -- I can do it once SB is updated.
